pkg/credentialprovider/plugin/plugin.go

```diff
@@ -269,6 +297,14 @@ func (p *pluginProvider) isImageAllowed(image string) bool {
 
 // getCachedCredentials returns a credentialprovider.DockerConfig if cached from the plugin.
 func (p *pluginProvider) getCachedCredentials(image string) (credentialprovider.DockerConfig, bool, error) {
+	// NewExpirationCache purges expired entries when List() is called
+	p.Lock()
+	if p.clock.Now().After(p.lastCachePurge.Add(cachePurgeInterval)) {
+		_ = p.cache.List()
```

Comment:
Going back to one of my older comments, relying on List() for cache purging seems a bit odd. I would at least add a comment here for why List() is called. 

Alternatively as a follow-up, it might make sense to just write a simple cache implementation internal to this package (or use a caching lib from another package) that handles this use-case a bit better.

Reply:
I have added a comment on Line no. 300, Updated it to be more explicit.
With this cache there is no other way we have, on the side not We were using this cache with intree ones since long time without purging :)

Reply:
Also a reminder that this PR is only scoped for external traffic, I think we had some discussions on whether we should handle internal but I didn't want to implement that on the first pass.